test/suite/test_tiered15.py

```diff
@@ -0,0 +1,86 @@
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_tiered15.py
+#   Test the "type" configuration in session.create with tiered storage.
+
+from helper_tiered import TieredConfigMixin, tiered_storage_sources
+from wtscenario import make_scenarios
+import wiredtiger, wttest
+
+class test_tiered15(TieredConfigMixin, wttest.WiredTigerTestCase):
+    types = [
+        ('file', dict(type='file', tiered_err=False, non_tiered_err=False, tiered_errmsg=None, non_tiered_errmsg=None)),
+        ('table', dict(type='table', tiered_err=True, non_tiered_err=False, tiered_errmsg="/Operation not supported/", non_tiered_errmsg=None)),
+        ('tier', dict(type='tier', tiered_err=True, non_tiered_err=False, tiered_errmsg="/Operation not supported/", non_tiered_errmsg=None)),
+        ('tiered', dict(type='tiered', tiered_err=True, non_tiered_err=True, tiered_errmsg="/Operation not supported/", non_tiered_errmsg="/Invalid argument/")),
+        ('colgroup', dict(type='colgroup', tiered_err=True, non_tiered_err=True, tiered_errmsg="/Operation not supported/", non_tiered_errmsg=None)),
+        ('index', dict(type='index', tiered_err=True, non_tiered_err=True, tiered_errmsg="/Operation not supported/", non_tiered_errmsg="/Invalid argument/")),
+        ('lsm', dict(type='lsm', tiered_err=True, non_tiered_err=False, tiered_errmsg="/Operation not supported/", non_tiered_errmsg=None)),
+        ('backup', dict(type='backup', tiered_err=True, non_tiered_err=False, tiered_errmsg="/Operation not supported/", non_tiered_errmsg="/Operation not supported/")),
+    ]
+
+    # This is different to the self.is_tiered_scenario() value - that one configures tiered
+    # storage at a connection level but here we want to configure at the table level.
+    is_tiered_table = [
+        ('tiered_table', dict(is_tiered_table=True)),
+        ('nontiered_table', dict(is_tiered_table=False)),
+    ]
+
+    scenarios = make_scenarios(tiered_storage_sources, types)
+
+    def test_create_type_config(self):
+        if not self.is_tiered_scenario():
+            self.skipTest('The test should only test connections configured with tiered storage.')
+
+        tiered_status = "tiered" if self.is_tiered_table else "nontiered"
+        uri = "table:" + tiered_status + self.type
+
+        if self.is_tiered_table:
+            # Creating a tiered table with the type configuration set to file should succeed, and all other types
+            # should not succeed.
+            if self.type == "file":
+                self.session.create(uri, "type=" + self.type)
+            else:
```

Comment:
Ah, you already have the. conditional. You can just put the expected error message here and check for it.
```
msg = '/Operation not supported/'
self.assert...(..., msg)
```